tests/firedrake/regression/test_assemble_baseform.py

```diff
@@ -155,6 +155,7 @@ def test_zero_form(M, f, one):
     assert abs(zero_form - 0.5 * np.prod(f.ufl_shape)) < 1.0e-12
 
 
+@pytest.mark.xfail(reason="action(M, M) raises primal-dual TypeError")
```

Comment:
This suggests that this PR breaks things for users?

Reply:
I doubt anyone has done something like squaring a mass matrix to do something useful, and if they do they should stop, as that produces a mesh-dependent algorithm.

Reply:
Does that mean that this test should be deleted?

Reply:
The test is testing something else, but I think it is very hard to recreate on an object that makes mathematical sense.

Reply:
I have deleted this test, as the behaivor we were testing cannot be reproduced with `action(action(Form, Interpolator), Function)`